src/Package/Impl/DataInspect/Viewers/FunctionViewer.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.VisualStudio.R.Package.Repl;
+using Microsoft.VisualStudio.R.Package.Shell;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IObjectDetailsViewer))]
+    internal sealed class FunctionViewer : IObjectDetailsViewer {
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly IDebugObjectEvaluator _evaluator;
+
+        [ImportingConstructor]
+        public FunctionViewer(IRSessionProvider sessionProvider, IDebugObjectEvaluator evaluator) {
+            _sessionProvider = sessionProvider;
+            _evaluator = evaluator;
+        }
+
+        #region IObjectDetailsViewer
+        public ViewerCapabilities Capabilities => ViewerCapabilities.Function;
+
+        public bool CanView(DebugValueEvaluationResult evaluation) {
+            return evaluation != null && evaluation.Classes.Count == 1 && evaluation.Classes[0].EqualsOrdinal("function");
+        }
+
+        public async Task ViewAsync(string expression, string title) {
+            var evaluation = await _evaluator.EvaluateAsync(expression, DebugEvaluationResultFields.Expression) as DebugValueEvaluationResult;
+            if(evaluation == null) {
+                return;
+            }
+
+            string functionName = evaluation.Expression as string;
+            if (string.IsNullOrEmpty(functionName)) {
+                return;
+            }
+
+            var session = _sessionProvider.GetInteractiveWindowRSession();
+            string functionCode = null;
+            using (var e = await session.BeginEvaluationAsync()) {
+                var result = await e.EvaluateAsync(Invariant($"paste0(deparse({functionName}), collapse='\n')"), REvaluationKind.Normal);
+                if (result.ParseStatus == RParseStatus.OK && result.Error == null && result.StringResult != null) {
+                    functionCode = result.StringResult;
+                }
+            }
+
+            if (!string.IsNullOrEmpty(functionCode)) {
+                var formatter = new RFormatter(REditorSettings.FormatOptions);
+                functionCode = formatter.Format(functionCode);
+
+                string name = (!string.IsNullOrEmpty(title) && title.IndexOfAny(Path.GetInvalidFileNameChars()) < 0) ? title : functionName;
+                string fileName = "~" + name;
+                string tempFile = Path.Combine(Path.GetTempPath(), Path.ChangeExtension(fileName, ".r"));
+                try {
+                    if (File.Exists(tempFile)) {
+                        File.Delete(tempFile);
+                    }
+
+                    using (var sw = new StreamWriter(tempFile)) {
+                        sw.Write(functionCode);
+                    }
+
+                    VsAppShell.Current.DispatchOnUIThread(() => {
+                        var dte = VsAppShell.Current.GetGlobalService<DTE>();
+                        dte.ItemOperations.OpenFile(tempFile);
```

Comment:
No, editor does not lock files. Yeah, we can look into transients at some point, not sure if they are supported in all scenarios.